### PR TITLE
Update git ssh url pattern

### PR DESCRIPTION
**Pull request by matthias-p**
### Motivation and context

The changes allow for an ssh git repo url to start with ssh://git@... .  It also allows user expansion in the url which is used by e.g. Bitbucket.
Issues this solves: #2737 

### How has this been tested?

This has been tested manually with git urls including and excluding the ssh:// prefix

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nmanovic** [comment]
@matthias-p , thanks for the PR. Could you please resolve conflicts?

**Comment by matthias-p** [comment]
Thank you, I have resolved them.

File: cvat/apps/dataset_repo/dataset_repo.py [hunk 1]
```diff
@@ -94,9 +94,9 @@ def _parse_url(self):
                 host = http_match.group(1)
                 repos = http_match.group(2)[1:]
             elif ssh_match:
-                user = ssh_match.group(1)
-                host = ssh_match.group(2)
-                repos = "{}{}".format(ssh_match.group(3), ssh_match.group(4))
+                user = ssh_match.group(1) + ssh_match.group(2) if ssh_match.groups(1) is not None else ssh_match.group(2)
```
Comment (nmanovic) [inline cvat/apps/dataset_repo/dataset_repo.py:97]: @matthias-p , I believe you have a typo here: `ssh_match.groups`

**Comment by nmanovic** [comment]
Linter was failing due to problems with popen. It isn't a problem of the patch. Let's ignore it.

File: cvat/apps/dataset_repo/dataset_repo.py (hunk 1 above)
Comment (matthias-p) [reply cvat/apps/dataset_repo/dataset_repo.py]: @nmanovic , indeed, thank you, I fixed it.

**Review by nmanovic — APPROVED** [review]
LGTM. Thanks for the contribution.

**Pull request merged**